Containerfile

```diff
@@ -8,25 +8,25 @@ RUN dnf -y install --setopt=install_weak_deps=False --setopt=tsdocs=False \
     sshpass \
     python3 \
     python3-pip \
+    python3-wheel \
+    pkgconf-pkg-config \
     && dnf clean all
 
-# Install ansible and ansible-lint
-RUN pip3 install --no-cache-dir \
-    ansible \
-    ansible-lint \
-    netaddr \
-    jmespath \
-    paramiko \
-    ncclient \
-    requests \
-    jira \
-    lxml \
-    junitparser
+# Copy python requirements file to eco-ci-cd folder
+COPY pip.txt .
+
+# Install python dependencies
+RUN python3 -m pip install --upgrade pip && \
+    python3 -m pip install --prefer-binary --no-cache-dir --no-compile -r pip.txt
+
+# Copy files affecting ansible-galaxy to eco-ci-cd folder
+COPY requirements.yml ansible.cfg .
+# Install galaxy requirements
+RUN ansible-galaxy collection install --no-cache --force --pre -r requirements.yml
 
-# Copy application files to eco-ci-cd folder
 COPY . .
-# Install requirements
-RUN ansible-galaxy collection install -r requirements.yml
+ENV \
+    ANSIBLE_COLLECTIONS_PATH=/eco-ci-cd/collections:/usr/local/lib/python3.9/site-packages
```

Comment:
I don't think we need this - hardcoding the python version is bad practice 
can we remove it ?